edgelet/edgelet-http/src/lib.rs

```diff
@@ -349,7 +348,19 @@ impl HyperExt for Http {
                 let cert_identity = Identity::from_pkcs12(&cert, "")
                     .context(ErrorKind::TlsIdentityCreationError)?;
 
+                let min_protocol_version = tls_params.as_ref().and_then(|params| {
+                    params
+                        .min_protocol_version
+                        .as_ref()
+                        .map(|protocol| match protocol {
+                            Protocol::Tls10 => native_tls::Protocol::Tlsv10,
+                            Protocol::Tls11 => native_tls::Protocol::Tlsv11,
+                            Protocol::Tls12 => native_tls::Protocol::Tlsv12,
+                        })
+                });
+
                 let tls_acceptor = TlsAcceptor::builder(cert_identity)
+                    .min_protocol_version(min_protocol_version)
```

Comment:
This is worse than before, I believe. Before this change, the default was TLS 1.0. With this change, the default is the lowest one that openssl supports, which could be SSL 2.0.

Reply:
Do you really need the min version to be `Option` ? You could just make it a required property and do `serde(default = "Tlsv10")` when deserializing it. (Not exactly that because it needs a path of a function, not an expr.)

Reply:
Why? if the user didn't put anything `min_protocol_version = None`; then flow as it was before.
If the user put something and we parsed it successfully, it will be either of TLS1.0, TLS1.1, TLS1.2.
Sorry, I didn't quite understand your concern.